src/validation.cpp

```diff
@@ -4707,7 +4707,8 @@ bool LoadMempool()
         }
         uint64_t num;
         file >> num;
-        while (num--) {
+        while (num) {
+            --num;
```

Comment:
How do this avoid overflow?

Reply:
It won't decrement when `num` is 0. The old code would stop the loop, but still decrement.